[TASK] Parse command line arguments with argparse module

BafDyce opened the pull request:
In order to 1) clean up the codebase a bit and 2) make it easier to add new
features in the future, parsing command line arguments is now done with the
argparse module.

Unfortunately it is not possible to add new lines to the help-messages,
therefore '|' is used to separate explanations of the different format-choices.
Also, apparently it is not possible to rename the additional parameters for the
"--grid" option.

This commit also removes all functions which are not used anymore because of
this addition (print_usage, abort_with_usage, etc.). Hopefully I didn't forget
any.

Tested on Linux Mint 17.1 Cinnamon 64-bit and LibreOffice 4.2.8.2

Commit:
[TASK] Parse command line arguments with argparse module

In order to 1) clean up the codebase a bit and 2) make it easier to add new
features in the future, parsing command line arguments is now done with the
argparse module.

Unfortunately it is not possible to add new lines to the help-messages,
therefore '|' is used to separate explanations of the different format-choices.
Also, apparently it is not possible to rename the additional parameters for the
"--grid" option.

This commit also removes all functions which are not used anymore because of
this addition (print_usage, abort_with_usage, etc.). Hopefully I didn't forget
any.

Tested on Linux Mint 17.1 Cinnamon 64-bit and LibreOffice 4.2.8.2

Dobiasd commented:
Great work! It looks much cleaner than before and simultaneously saves about 30 lines of code.

Just one small thing: At the moment the `--grid` option is not really optional but mandatory.

```
python3 img2xls.py libre images/lenna_64x64.png

Traceback (most recent call last):
  File "img2xls.py", line 183, in <module>
    sys.exit(main())
  File "img2xls.py", line 173, in main
    grid_gap_vert = args.grid[0]
TypeError: 'int' object is not subscriptable
```

Would you like to fix it for your commit?

BafDyce commented:
Huh? Well, seems like I forgot to test that, even though I was sure I also had a test run without the --grid argument.
However, I can confirm the error. I'll try to write a fix for it as soon as possible, but I probably won't have time for it today anymore.

Commit:
[FIX] Fixed crash when running without --grid option

Cause of the bug was that in case the --grid option was not specified, the
variable args.grid was assigned the integer value 0. Later args.grid was
accessed as if it was an array.

The fix: In case --grid is not specified, [0, 0] (an array consisting of two
zeros) is now assigned to args.grid.

Tested on Linux Mint 17.1 Cinnamon 64-bit and LibreOffice 4.2.8.2

Commit:
[TASK] Updated README

Included full help message, as produced by the program.

BafDyce commented:
So, another two commits.
First one fixes the crash, the second one updates the readme. (Includes the help output, as produced by the program itself. Also fixed the line "blabla.xls will be generated" to also include the specified format)

Dobiasd commented:
:thumbsup:
